HossainAraf/metrics-webapp
# React capstone project - Metrics webapp

A mobile web application

***Methods implemented:***

-  Chosse an API should allow us to get numeric values based on a parameter;
- Interactions:

**Home page**

When the page loads, the webapp shows a bunch of countries with their corresponding population.

When the user clicks (or taps) on a category item (A Country), the application navigates to the details page.

**Details page**

In the details page, the webapp retrieves data from the API to show detailed data (Currency, capital etc.) in the category.
-When the user clicks on the "Back" button (<), the user navigates to the home page.

Testing requirements


- Create unit tests for pure functions in Redux code.
- Create integration tests for the application using the React Testing Library.

- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- We should filter the data that we retrieve from the API using a Filter stateless component;
- Every page (the main page and the pages for each item) should have a unique route within the SPA.
- The project should be deployed and accessible online using [Render](https://render.com/), [Netlify](https://www.netlify.com/) or [GitHub Pages](https://pages.github.com/).
- Add a link to the online version of your app to your README file.
